### ISSUE TITLE
ForeignKey object returns null after relay.ClientIDMutation

### ISSUE THREAD

**Issue by jtl1990** [graphql-python/graphene-django#1062]
* **What is the current behavior?**

ForeignKey object returns null after `relay.ClientIDMutation` although the related object is successfully created.

```
{
  "data": {
    "createUser": {
      "user": {
        "address": null
      }
    }
  }
}
```

CreateUser Mutation code will look something similar to this
```
class CreateUser(relay.ClientIDMutation):
      class Input:
          address = Field(AddressInput, required=False)

      user = Field(User)

      @classmethod
      def mutate_and_get_payload(cls, root, info, **input):
          address_payload = input.get('address')
          address_obj = AddressModel.objects.create(**address_payload)

          user = UserModel.objects.create_user(address=address_obj)

          return CreateUser(user=user)
```

* **What is the expected behavior?**
```
{
  "data": {
    "createUser": {
      "user": {
        "address": {
          "address1": "123 Testing Address St."
        }
      }
    }
  }
}
```

* **Please tell us about your environment:**
  
  - Version: 2.13.0
  - Platform: django 3.1.2

**Comment by zbyte64** [comment]
Three questions:

* is `Field` from `relay.Node.Field`?
* Does `User` have:

```
class Meta:
        interfaces = (relay.Node, )
```

* does this work with `graphene.Mutation` in place of `ClientIDMutation`?

But I do think you're right about expected behavior...

**Comment by toboklee** [comment]
@zbyte64 

> is Field from relay.Node.Field?

No, it's `graphene.Field`. `relay.Node.Field` would require an ID... not sure if it's possible to be used in this case.
Looks like the examples from official doc is also using `graphene.Field`
https://docs.graphene-python.org/en/latest/relay/mutations/

> Does User have:
> class Meta:
        interfaces = (relay.Node, )

Yup. 

> does this work with graphene.Mutation in place of ClientIDMutation?

No :(, I changed the code and tried with `graphene.Mutation` and it's the same as current behaviour.

**Comment by zbyte64** [comment]
if `relay.Node.Field` is used to link with your `User` then it should work as `User` implements id and `relay.Node`. It's not that the parent object needs an id, its the object we are referring to that needs it.
